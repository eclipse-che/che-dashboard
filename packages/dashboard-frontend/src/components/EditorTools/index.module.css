/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

.editorTools {
  height: 20px;
  margin: 10px 0 5px;
}

.editorTools > * {
  float: right;
  height: inherit;
  font-size: small;
  user-select: none;
}

.editorTools a {
  text-decoration: none;
}

.editorTools button {
  padding: 0;
  font-size: inherit;
  outline: none;
}

.editorTools svg {
  margin-right: 8px;
}
