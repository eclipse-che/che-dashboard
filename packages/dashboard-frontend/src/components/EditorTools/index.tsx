/*
 * Copyright (c) 2018-2023 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { ApplicationId, helpers } from '@eclipse-che/common';
import { AlertVariant, Button, Divider, Flex, FlexItem } from '@patternfly/react-core';
import {
  CompressIcon,
  CopyIcon,
  DownloadIcon,
  ExpandIcon,
  ExternalLinkSquareAltIcon,
} from '@patternfly/react-icons';
import React from 'react';
import CopyToClipboard from 'react-copy-to-clipboard';
import { ConnectedProps, connect } from 'react-redux';
import { ToggleBarsContext } from '../../contexts/ToggleBars';
import { lazyInject } from '../../inversify.config';
import { AppAlerts } from '../../services/alerts/appAlerts';
import devfileApi, { isDevWorkspace, isDevfileV2 } from '../../services/devfileApi';
import stringify from '../../services/helpers/editor';
import { AlertItem } from '../../services/helpers/types';
import { AppState } from '../../store';
import { actionCreators } from '../../store/BannerAlert';
import { selectApplications } from '../../store/ClusterInfo/selectors';
import styles from './index.module.css';
import { WorkspaceAdapter } from '../../services/workspace-adapter';

type Props = MappedProps & {
  devfileOrDevWorkspace: devfileApi.DevWorkspace | devfileApi.Devfile;
  handleExpand: (isExpand: boolean) => void;
};

type State = {
  copied?: boolean;
  isExpanded: boolean;
  contentText: string;
  contentBlobUrl: string;
};

class EditorTools extends React.PureComponent<Props, State> {
  static contextType = ToggleBarsContext;
  readonly context: React.ContextType<typeof ToggleBarsContext>;

  private copiedTimer: number | undefined;

  @lazyInject(AppAlerts)
  private readonly appAlerts: AppAlerts;

  constructor(props: Props) {
    super(props);

    this.state = {
      isExpanded: false,
      contentText: '',
      contentBlobUrl: '',
    };
  }

  public componentDidMount(): void {
    this.init();
  }

  public componentDidUpdate(): void {
    this.init();
  }

  private init() {
    const { devfileOrDevWorkspace } = this.props;
    try {
      const contentText = stringify(devfileOrDevWorkspace);
      if (contentText !== this.state.contentText) {
        const contentBlobUrl = URL.createObjectURL(
          new Blob([contentText], { type: 'application/x-yaml' }),
        );
        this.setState({ contentText, contentBlobUrl });
      }
    } catch (e) {
      this.showAlert({
        key: 'editor-tools-create blob-url-fails',
        variant: AlertVariant.danger,
        title: helpers.errors.getMessage(e),
      });
    }
  }

  private showAlert(alert: AlertItem): void {
    this.appAlerts.showAlert(alert);
  }

  private handleExpand(): void {
    if (this.state.isExpanded) {
      this.context.showAll();
      const isExpanded = false;
      this.setState({ isExpanded });
      this.props.handleExpand(isExpanded);
    } else {
      this.context.hideAll();
      const isExpanded = true;
      this.setState({ isExpanded });
      this.props.handleExpand(isExpanded);
    }
  }

  private onCopyToClipboard(): void {
    this.setState({ copied: true });
    if (this.copiedTimer) {
      clearTimeout(this.copiedTimer);
    }
    this.copiedTimer = window.setTimeout(() => {
      this.setState({ copied: false });
    }, 3000);
  }

  private buildOpenShiftConsoleItem(): React.ReactElement | undefined {
    const { applications, devfileOrDevWorkspace } = this.props;
    const clusterConsole = applications.find(app => app.id === ApplicationId.CLUSTER_CONSOLE);

    if (isDevfileV2(devfileOrDevWorkspace)) {
      return;
    }
    const devWorkspace = devfileOrDevWorkspace;

    if (!clusterConsole) {
      return;
    }

    const devWorkspaceOpenShiftConsoleUrl = WorkspaceAdapter.buildClusterConsoleUrl(
      devWorkspace,
      clusterConsole.url,
    );

    return (
      <>
        <FlexItem>
          <Button
            className={styles.button}
            component="a"
            variant="link"
            href={devWorkspaceOpenShiftConsoleUrl}
            target="_blank"
            icon={<ExternalLinkSquareAltIcon />}
          >
            {clusterConsole.title}
          </Button>
        </FlexItem>
        <Divider isVertical />
      </>
    );
  }

  public render(): React.ReactElement {
    const { devfileOrDevWorkspace } = this.props;
    const { contentText, contentBlobUrl, isExpanded, copied } = this.state;

    const { name } = devfileOrDevWorkspace.metadata;

    const openshiftConsoleItem = this.buildOpenShiftConsoleItem();

    return (
      <div className={styles.editorTools}>
        <Flex>
          {openshiftConsoleItem}
          <FlexItem>
            <CopyToClipboard text={contentText} onCopy={() => this.onCopyToClipboard()}>
              <Button variant="link" className={styles.button}>
                <CopyIcon />
                {copied ? 'Copied' : 'Copy to clipboard'}
              </Button>
            </CopyToClipboard>
          </FlexItem>
          <Divider isVertical />
          <FlexItem>
            <a
              className={styles.button}
              download={`${name}.${
                isDevWorkspace(devfileOrDevWorkspace) ? 'workspace' : 'devfile'
              }.yaml`}
              href={contentBlobUrl}
            >
              <DownloadIcon />
              Download
            </a>
          </FlexItem>
          <Divider isVertical />
          <FlexItem>
            <Button className={styles.button} variant="link" onClick={() => this.handleExpand()}>
              {isExpanded ? (
                <React.Fragment>
                  <CompressIcon />
                  Compress
                </React.Fragment>
              ) : (
                <React.Fragment>
                  <ExpandIcon />
                  Expand
                </React.Fragment>
              )}
            </Button>
          </FlexItem>
        </Flex>
      </div>
    );
  }
}

const mapStateToProps = (state: AppState) => ({
  applications: selectApplications(state),
});

const connector = connect(mapStateToProps, actionCreators);

type MappedProps = ConnectedProps<typeof connector>;

export default connector(EditorTools);
