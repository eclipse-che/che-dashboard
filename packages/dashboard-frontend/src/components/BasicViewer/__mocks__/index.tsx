/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/BasicViewer';

export class BasicViewer extends React.PureComponent<Props> {
  public render(): React.ReactNode {
    const { id, value } = this.props;

    return (
      <div>
        Mock Basic Viewer
        <span data-testid="basic-viewer-id">{id}</span>
        <span data-testid="basic-viewer-value">{value}</span>
      </div>
    );
  }
}
