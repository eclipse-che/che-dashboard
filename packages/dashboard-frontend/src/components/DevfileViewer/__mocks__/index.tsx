/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/DevfileViewer';

export class DevfileViewer extends React.PureComponent<Props> {
  public render(): React.ReactNode {
    const { isActive, isExpanded, value } = this.props;

    return (
      <div>
        Mock Devfile Viewer
        <span data-testid="devfile-viewer-is-active">{isActive.toString()}</span>
        <span data-testid="devfile-viewer-is-expanded">{isExpanded.toString()}</span>
        <span data-testid="devfile-viewer-value">{value}</span>
      </div>
    );
  }
}
