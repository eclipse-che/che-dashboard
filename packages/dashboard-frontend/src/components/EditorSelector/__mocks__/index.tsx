/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/EditorSelector';

export default class EditorSelector extends React.PureComponent<Props> {
  render() {
    const { onSelect } = this.props;
    return (
      <div data-testid="editor-selector">
        Editor Selector
        <button onClick={() => onSelect('some/editor/id', 'custom-editor-image')}>
          Select Editor
        </button>
      </div>
    );
  }
}
