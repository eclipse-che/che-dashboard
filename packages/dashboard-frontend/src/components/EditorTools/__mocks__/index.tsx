/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

// mock for the EditorTools component
import React from 'react';

import { Props } from '@/components/EditorTools';

export default class EditorTools extends React.PureComponent<Props> {
  public render(): React.ReactNode {
    const { handleExpand } = this.props;
    return (
      <div>
        Mock Editor Tools
        <button onClick={() => handleExpand(true)}>Expand Editor</button>
      </div>
    );
  }
}
