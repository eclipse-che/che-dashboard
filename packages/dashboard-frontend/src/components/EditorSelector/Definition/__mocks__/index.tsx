/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/EditorSelector/Definition';

export class EditorDefinition extends React.PureComponent<Props> {
  public render() {
    const { editorDefinition, editorImage, onChange } = this.props;

    return (
      <div data-testid="editor-definition-component">
        <div>Editor Definition</div>
        <div data-testid="editor-definition">{editorDefinition}</div>
        <div data-testid="editor-image">{editorImage}</div>
        <button onClick={() => onChange('some/editor/id', 'editor-image')}>
          Editor Definition Change
        </button>
      </div>
    );
  }
}
