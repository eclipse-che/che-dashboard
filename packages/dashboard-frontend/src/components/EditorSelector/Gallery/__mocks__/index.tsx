/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/EditorSelector/Gallery';

export class EditorGallery extends React.PureComponent<Props> {
  public render() {
    const { defaultEditorId, selectedEditorId, onSelect } = this.props;

    return (
      <div data-testid="editor-gallery-component">
        <div>Editor Gallery</div>
        <div data-testid="default-editor-id">{defaultEditorId}</div>
        <div data-testid="selected-editor-id">{selectedEditorId}</div>
        <button onClick={() => onSelect('che-incubator/che-code/latest')}>Select Editor</button>
      </div>
    );
  }
}
