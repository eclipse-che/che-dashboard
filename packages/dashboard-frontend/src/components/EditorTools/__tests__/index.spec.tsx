/*
 * Copyright (c) 2018-2023 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import getComponentRenderer, { screen } from '../../../services/__mocks__/getComponentRenderer';
import { Provider } from 'react-redux';
import EditorTools from '..';
import { Store } from 'redux';
import { FakeStoreBuilder } from '../../../store/__mocks__/storeBuilder';
import devfileApi from '../../../services/devfileApi';
import { ApplicationId } from '@eclipse-che/common';

jest.mock('../../../contexts/ToggleBars');

const mockClipboard = jest.fn();
jest.mock('react-copy-to-clipboard', () => {
  return {
    __esModule: true,
    default: (props: any) => {
      return (
        <button
          onClick={() => {
            mockClipboard(props.text);
            props.onCopy();
          }}
        >
          Copy to clipboard
        </button>
      );
    },
  };
});

const { renderComponent, createSnapshot } = getComponentRenderer(getComponent);

const mockOnExpand = jest.fn();
let store: Store;

describe('EditorTools', () => {
  const clusterConsole = {
    id: ApplicationId.CLUSTER_CONSOLE,
    url: 'https://console-url',
    icon: 'https://console-icon-url',
    title: 'Cluster console',
  };

  beforeEach(() => {
    store = new FakeStoreBuilder()
      .withClusterInfo({
        applications: [clusterConsole],
      })
      .build();

    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  describe('Devfile', () => {
    let devfile: devfileApi.Devfile;

    beforeEach(() => {
      devfile = {
        schemaVersion: '2.1.0',
        metadata: {
          name: 'my-project',
          namespace: 'user-che',
        },
      };
    });

    test('snapshot', () => {
      const snapshot = createSnapshot(devfile);
      expect(snapshot.toJSON()).toMatchSnapshot();
    });

    test('expand and compress', () => {
      renderComponent(devfile);

      /* expand the editor */

      const expandButtonName = 'Expand';
      expect(screen.getByRole('button', { name: expandButtonName })).toBeTruthy;

      screen.getByRole('button', { name: expandButtonName }).click();
      expect(mockOnExpand).toHaveBeenCalledWith(true);

      /* compress the editor */

      const compressButtonName = 'Compress';
      expect(screen.getByRole('button', { name: compressButtonName })).toBeTruthy;

      screen.getByRole('button', { name: compressButtonName }).click();
      expect(mockOnExpand).toHaveBeenCalledWith(false);
    });

    test('copy to clipboard', () => {
      const mockCreateObjectURL = jest.fn().mockReturnValue('blob-url');
      URL.createObjectURL = mockCreateObjectURL;

      renderComponent(devfile);

      const copyButtonName = 'Copy to clipboard';
      expect(screen.queryByRole('button', { name: copyButtonName })).toBeTruthy;

      screen.getByRole('button', { name: copyButtonName }).click();

      expect(mockClipboard).toHaveBeenCalledWith(
        'schemaVersion: 2.1.0\nmetadata:\n  name: my-project\n  namespace: user-che\n',
      );

      /* 'Copy to clipboard' should be hidden for a while */

      expect(screen.queryByRole('button', { name: copyButtonName })).toBeFalsy;

      const copyButtonNameAfter = 'Copied';
      expect(screen.queryByRole('button', { name: copyButtonNameAfter })).toBeTruthy;

      /* 'Copy to clipboard' should re-appear after 3000ms */

      jest.advanceTimersByTime(4000);
      expect(screen.queryByRole('button', { name: copyButtonName })).toBeTruthy;
      expect(screen.queryByRole('button', { name: copyButtonNameAfter })).toBeFalsy;
    });
  });

  describe('DevWorkspace', () => {
    let devWorkspace: devfileApi.DevWorkspace;

    beforeEach(() => {
      devWorkspace = {
        apiVersion: '1.0.0',
        metadata: {
          name: 'my-project',
          namespace: 'user-che',
          labels: {},
          uid: '123',
        },
        kind: 'DevWorkspace',
        spec: {
          template: {},
          started: true,
        },
      };
    });

    test('snapshot', () => {
      const snapshot = createSnapshot(devWorkspace);
      expect(snapshot.toJSON()).toMatchSnapshot();
    });

    test('Cluster Console', () => {
      renderComponent(devWorkspace);

      const clusterConsoleButton = screen.getByRole('link', { name: clusterConsole.title });

      expect(clusterConsoleButton.textContent).toEqual(clusterConsole.title);
    });
  });
});

function getComponent(devfileOrDevWorkspace: devfileApi.Devfile | devfileApi.DevWorkspace) {
  return (
    <Provider store={store}>
      <EditorTools devfileOrDevWorkspace={devfileOrDevWorkspace} handleExpand={mockOnExpand} />
    </Provider>
  );
}
