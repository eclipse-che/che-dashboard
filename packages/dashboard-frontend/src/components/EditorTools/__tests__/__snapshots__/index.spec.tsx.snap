// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorTools DevWorkspace snapshot 1`] = `
<div
  className="editorTools"
>
  <div
    className="pf-l-flex"
  >
    <div
      className=""
    >
      <a
        aria-disabled={false}
        aria-label={null}
        className="pf-c-button pf-m-link button"
        data-ouia-component-id="OUIA-Generated-Button-link-4"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe={true}
        disabled={null}
        href="https://console-url/k8s/ns/user-che/1.0.0~DevWorkspace/my-project"
        role={null}
        target="_blank"
        type={null}
      >
        <span
          className="pf-c-button__icon pf-m-start"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            fill="currentColor"
            height="1em"
            role="img"
            style={
              {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z"
            />
          </svg>
        </span>
        Cluster console
      </a>
    </div>
    <hr
      className="pf-c-divider pf-m-vertical"
    />
    <div
      className=""
    >
      <button
        onClick={[Function]}
      >
        Copy to clipboard
      </button>
    </div>
    <hr
      className="pf-c-divider pf-m-vertical"
    />
    <div
      className=""
    >
      <a
        className="button"
        download="my-project.workspace.yaml"
        href="blob-url"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
          />
        </svg>
        Download
      </a>
    </div>
    <hr
      className="pf-c-divider pf-m-vertical"
    />
    <div
      className=""
    >
      <button
        aria-disabled={false}
        aria-label={null}
        className="pf-c-button pf-m-link button"
        data-ouia-component-id="OUIA-Generated-Button-link-5"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe={true}
        disabled={false}
        onClick={[Function]}
        role={null}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 448 512"
          width="1em"
        >
          <path
            d="M0 180V56c0-13.3 10.7-24 24-24h124c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H64v84c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12zM288 44v40c0 6.6 5.4 12 12 12h84v84c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12V56c0-13.3-10.7-24-24-24H300c-6.6 0-12 5.4-12 12zm148 276h-40c-6.6 0-12 5.4-12 12v84h-84c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h124c13.3 0 24-10.7 24-24V332c0-6.6-5.4-12-12-12zM160 468v-40c0-6.6-5.4-12-12-12H64v-84c0-6.6-5.4-12-12-12H12c-6.6 0-12 5.4-12 12v124c0 13.3 10.7 24 24 24h124c6.6 0 12-5.4 12-12z"
          />
        </svg>
        Expand
      </button>
    </div>
  </div>
</div>
`;

exports[`EditorTools Devfile snapshot 1`] = `
<div
  className="editorTools"
>
  <div
    className="pf-l-flex"
  >
    <div
      className=""
    >
      <button
        onClick={[Function]}
      >
        Copy to clipboard
      </button>
    </div>
    <hr
      className="pf-c-divider pf-m-vertical"
    />
    <div
      className=""
    >
      <a
        className="button"
        download="my-project.devfile.yaml"
        href=""
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
          />
        </svg>
        Download
      </a>
    </div>
    <hr
      className="pf-c-divider pf-m-vertical"
    />
    <div
      className=""
    >
      <button
        aria-disabled={false}
        aria-label={null}
        className="pf-c-button pf-m-link button"
        data-ouia-component-id="OUIA-Generated-Button-link-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe={true}
        disabled={false}
        onClick={[Function]}
        role={null}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 448 512"
          width="1em"
        >
          <path
            d="M0 180V56c0-13.3 10.7-24 24-24h124c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H64v84c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12zM288 44v40c0 6.6 5.4 12 12 12h84v84c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12V56c0-13.3-10.7-24-24-24H300c-6.6 0-12 5.4-12 12zm148 276h-40c-6.6 0-12 5.4-12 12v84h-84c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h124c13.3 0 24-10.7 24-24V332c0-6.6-5.4-12-12-12zM160 468v-40c0-6.6-5.4-12-12-12H64v-84c0-6.6-5.4-12-12-12H12c-6.6 0-12 5.4-12 12v124c0 13.3 10.7 24 24 24h124c6.6 0 12-5.4 12-12z"
          />
        </svg>
        Expand
      </button>
    </div>
  </div>
</div>
`;
